### PR TITLE
Alt support to OApp, OFT, OFTAdapter, and TestHelper

### PR DESCRIPTION
**Pull request by shankars99**
Added: 
1. `packages/oapp-evm-alt`
2. `packages/oft-evm-alt`

Modified: 
1. TestHelper to contain EndpointV2Alt support.

Todo: 
1. Once merged create `examples/oft-evm-alt`

### CODE REVIEW

File: packages/oapp-alt-evm/CHANGELOG.md
Comment (janjakubnanista) [inline packages/oapp-alt-evm/CHANGELOG.md:1]: Please delete

File: packages/oapp-alt-evm/README.md
Comment (janjakubnanista) [inline packages/oapp-alt-evm/README.md:1]: Please update the package name

File: packages/oapp-alt-evm/contracts/oapp/OAppCore.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/OAppCore.sol:1]: This whole file is copy-pasted. Please use the existing file from`oapp-evm`

File: packages/oapp-alt-evm/contracts/oapp/OAppReceiver.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/OAppReceiver.sol:1]: This whole file is copy-pasted. Please use the existing file from`oapp-evm`

File: packages/oapp-alt-evm/contracts/oapp/OAppSenderAlt.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/OAppSenderAlt.sol:1]: The `OAppSender` from `oapp-evm` should allow you to extend it & only overwrite the methods you need

File: packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppComposer.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppComposer.sol:1]: Please delete this in favour of `oapp-evm`

File: packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppCore.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppCore.sol:1]: Please delete in favour of `oapp-evm`

File: packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppMsgInspector.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppMsgInspector.sol:1]: Same as above

File: packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppOptionsType3.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppOptionsType3.sol:1]: Same as above

File: packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppReceiver.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppReceiver.sol:1]: Same as above

File: packages/oapp-alt-evm/contracts/oapp/libs/OAppOptionsType3.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/libs/OAppOptionsType3.sol:1]: Same as above

File: packages/oapp-alt-evm/contracts/oapp/libs/OptionsBuilder.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/libs/OptionsBuilder.sol:1]: Same as above

File: packages/oapp-alt-evm/contracts/oapp/utils/RateLimiter.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/utils/RateLimiter.sol:1]: Same as above

File: packages/oapp-alt-evm/test/OAppAlt.t.sol [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import { ERC20Mock } from "./mocks/ERC20Mock.sol";
+import { MyOAppAlt } from "./mocks/OAppAltMock.sol";
+import { TestHelperOz5 } from "@layerzerolabs/test-devtools-evm-foundry/contracts/TestHelperOz5.sol";
+import { MessagingParams, MessagingReceipt } from "@layerzerolabs/lz-evm-protocol-v2/contracts/interfaces/ILayerZeroEndpointV2.sol";
+import { OptionsBuilder } from "../contracts/oapp/libs/OptionsBuilder.sol";
+import { console } from "forge-std/console.sol";
+
+contract EndpointV2AltTest is TestHelperOz5 {
```
Comment (janjakubnanista) [inline packages/oapp-alt-evm/test/OAppAlt.t.sol:13]: `OAppAltTest`

File: packages/oapp-alt-evm/test/mocks/ERC20Mock.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/test/mocks/ERC20Mock.sol:1]: Please use the mock from `oapp-evm`

File: packages/oft-alt-evm/CHANGELOG.md
Comment (janjakubnanista) [inline packages/oft-alt-evm/CHANGELOG.md:1]: Please delete

File: packages/oft-alt-evm/README.md
Comment (janjakubnanista) [inline packages/oft-alt-evm/README.md:1]: Please update the package name

File: packages/oft-alt-evm/test/mocks/OFTComposerMock.sol
Comment (janjakubnanista) [inline packages/oft-alt-evm/test/mocks/OFTComposerMock.sol:1]: Please use the `oft-evm` version

File: packages/oft-alt-evm/test/mocks/OFTInspectorMock.sol
Comment (janjakubnanista) [inline packages/oft-alt-evm/test/mocks/OFTInspectorMock.sol:1]: Please use the `oft-evm` version

File: packages/oft-alt-evm/test/mocks/ERC20Mock.sol
Comment (janjakubnanista) [inline packages/oft-alt-evm/test/mocks/ERC20Mock.sol:1]: Please use the `oft-evm` version (or `oapp-evm` even better)

File: packages/oft-alt-evm/test/lib/OFTMockCodec.sol
Comment (janjakubnanista) [inline packages/oft-alt-evm/test/lib/OFTMockCodec.sol:1]: Please use the oft-evm version

File: packages/oft-alt-evm/test/OFTAlt.t.sol [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+import { OFTAdapterAlt } from "../contracts/OFTAdapterAlt.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+import { OFTMockCodec } from "./lib/OFTMockCodec.sol";
+import { OFTAdapterMockCodec } from "./lib/OFTAdapterMockCodec.sol";
+
+import { console } from "forge-std/console.sol";
+import { TestHelperOz5 } from "@layerzerolabs/test-devtools-evm-foundry/contracts/TestHelperOz5.sol";
+
+contract OFTTest is TestHelperOz5 {
```
Comment (janjakubnanista) [inline packages/oft-alt-evm/test/OFTAlt.t.sol:32]: `OFTAltTest`

File: packages/oft-alt-evm/test/Fee.t.sol
Comment (janjakubnanista) [inline packages/oft-alt-evm/test/Fee.t.sol:1]: Please remove in favour of the test in `oft-evm`

File: packages/oft-alt-evm/contracts/libs/OFTMsgCodec.sol
Comment (janjakubnanista) [inline packages/oft-alt-evm/contracts/libs/OFTMsgCodec.sol:1]: Please remove in favour of `oft-evm` version

File: packages/oft-alt-evm/contracts/libs/OFTComposeMsgCodec.sol
Comment (janjakubnanista) [inline packages/oft-alt-evm/contracts/libs/OFTComposeMsgCodec.sol:1]: Please remove in favour of `oft-evm` version

File: packages/oft-alt-evm/contracts/interfaces/IOFT.sol
Comment (janjakubnanista) [inline packages/oft-alt-evm/contracts/interfaces/IOFT.sol:1]: Please remove

File: packages/oft-alt-evm/contracts/interfaces/IFee.sol
Comment (janjakubnanista) [inline packages/oft-alt-evm/contracts/interfaces/IFee.sol:1]: Please remove

File: packages/oft-alt-evm/contracts/OFTAltCore.sol
Comment (janjakubnanista) [inline packages/oft-alt-evm/contracts/OFTAltCore.sol:1]: Please remove in favour of `oft-evm` version

File: packages/oft-alt-evm/contracts/Fee.sol
Comment (janjakubnanista) [inline packages/oft-alt-evm/contracts/Fee.sol:1]: Please remove in favour of `oft-evm` version

**Review by janjakubnanista — CHANGES_REQUESTED** [review]

File: packages/oapp-alt-evm/contracts/oapp/OAppCore.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/OAppCore.sol:1]: Please remove in favour of `oapp-evm` version

File: packages/oapp-alt-evm/contracts/oapp/OAppReceiver.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/OAppReceiver.sol:1]: Please remove in favour of `oapp-evm` version

File: packages/oapp-alt-evm/contracts/oapp/OAppSenderAlt.sol [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+import { SafeERC20, IERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { MessagingParams, MessagingFee, MessagingReceipt } from "@layerzerolabs/lz-evm-protocol-v2/contracts/interfaces/ILayerZeroEndpointV2.sol";
+import { OAppCore } from "./OAppCore.sol";
+
+/**
+ * @title OAppSender
+ * @dev Abstract contract implementing the OAppSender functionality for sending messages to a LayerZero endpoint.
+ */
+abstract contract OAppSenderAlt is OAppCore {
```
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/OAppSenderAlt.sol:13]: Please use the base version from `oapp-evm` and override what you need, I think that should be possible

File: packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppComposer.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppComposer.sol:1]: Please remove in favour of `oapp-evm` version

File: packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppCore.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppCore.sol:1]: Please remove in favour of `oapp-evm` version

File: packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppMsgInspector.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppMsgInspector.sol:1]: Please remove in favour of `oapp-evm` version

File: packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppOptionsType3.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppOptionsType3.sol:1]: Please remove in favour of `oapp-evm` version

File: packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppReceiver.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/interfaces/IOAppReceiver.sol:1]: Please remove in favour of `oapp-evm` version

File: packages/oapp-alt-evm/contracts/oapp/libs/OAppOptionsType3.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/libs/OAppOptionsType3.sol:1]: Please remove in favour of `oapp-evm` version

File: packages/oapp-alt-evm/contracts/oapp/libs/OptionsBuilder.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/libs/OptionsBuilder.sol:1]: Please remove in favour of `oapp-evm` version

File: packages/oapp-alt-evm/contracts/oapp/utils/RateLimiter.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/contracts/oapp/utils/RateLimiter.sol:1]: Please remove in favour of `oapp-evm` version

File: packages/oapp-alt-evm/test/mocks/ERC20Mock.sol
Comment (janjakubnanista) [inline packages/oapp-alt-evm/test/mocks/ERC20Mock.sol:1]: Please remove in favour of `oapp-evm` version

File: packages/oft-alt-evm/CHANGELOG.md
Comment (janjakubnanista) [inline packages/oft-alt-evm/CHANGELOG.md:1]: Please delete

File: packages/oft-alt-evm/README.md
Comment (janjakubnanista) [inline packages/oft-alt-evm/README.md:1]: Please update package names

File: packages/oft-alt-evm/contracts/Fee.sol
Comment (janjakubnanista) [inline packages/oft-alt-evm/contracts/Fee.sol:1]: Please remove in favour of `oft-evm` version

**Review by janjakubnanista — CHANGES_REQUESTED** [review]
Please extrapolate the comments to the whole contents of the PR.

I would also split the two packages into two separate PRs

File: packages/oapp-alt-evm/test/mocks/ERC20Mock.sol
Comment (shankars99) [reply packages/oapp-alt-evm/test/mocks/ERC20Mock.sol:1]: https://github.com/LayerZero-Labs/devtools/tree/main/packages/oapp-evm/test/mocks

doesnt have test cases or ERC20Mock

**Pull request closed without merging**